src/pretix/plugins/paypal/static/pretixplugins/paypal/pretix-paypal.js

```diff
@@ -0,0 +1,280 @@
+            pretixpaypal.continue_button = $('.checkout-button-row').closest("form").find(".checkout-button-row .btn-primary");
+            pretixpaypal.continue_button.closest('div').append('<div id="paypal-button-container"></div>');
+            pretixpaypal.additional_disabled_funding = $.trim($("#paypal_disable_funding").html());
+            pretixpaypal.additional_enabled_funding = $.trim($("#paypal_enable_funding").html());
+            pretixpaypal.paypage = Boolean($('#paypal-button-container').data('paypage'));
+            pretixpaypal.order_id = $.trim($("#paypal_oid").html());
+            pretixpaypal.currency = $("body").attr("data-currency");
+        }
+
+        pretixpaypal.continue_button.prop("disabled", true);
+
+        let sdk_url = 'https://www.paypal.com/sdk/js' +
+            '?client-id=' + pretixpaypal.client_id +
+            '&components=buttons,funding-eligibility' +
+            '&currency=' + pretixpaypal.currency;
+
+        if (pretixpaypal.merchant_id) {
+            sdk_url += '&merchant-id=' + pretixpaypal.merchant_id;
+        }
+
+        if (pretixpaypal.additional_disabled_funding) {
+            sdk_url += '&disable-funding=' + [pretixpaypal.additional_disabled_funding].filter(Boolean).join(',');
+        }
+
+        if (pretixpaypal.additional_enabled_funding) {
+            sdk_url += '&enable-funding=' + [pretixpaypal.additional_enabled_funding].filter(Boolean).join(',');
+        }
+
+        if (pretixpaypal.debug_buyer_country) {
+            sdk_url += '&buyer-country=' + pretixpaypal.debug_buyer_country;
+        }
+
+        let ppscript = document.createElement('script');
+        let ready = false;
+        let head = document.getElementsByTagName("head")[0];
+        ppscript.setAttribute('src', sdk_url);
+        ppscript.setAttribute('data-csp-nonce', $.trim($("#csp_nonce").html()));
+        ppscript.setAttribute('data-page-type', 'checkout');
+        ppscript.setAttribute('data-partner-attribution-id', 'ramiioGmbH_Cart_PPCP');
+        document.head.appendChild(ppscript);
+
+        ppscript.onload = ppscript.onreadystatechange = function () {
+            if (!ready && (!this.readyState || this.readyState === "loaded" || this.readyState === "complete")) {
+                ready = true;
+
+                pretixpaypal.paypal = paypal;
+
+                // Handle memory leak in IE
+                ppscript.onload = ppscript.onreadystatechange = null;
+                if (head && ppscript.parentNode) {
+                    head.removeChild(ppscript);
+                }
+            }
+        };
+    },
+
+    ready: function () {
+        if ($("input[name=payment][value=paypal_apm]").length > 0) {
+            pretixpaypal.renderAPMs();
+        }
+
+        $("input[name=payment][value^='paypal']").change(function () {
+            pretixpaypal.renderButton($(this).val());
+        });
+
+        $("input[name=payment]").not("[value^='paypal']").change(function () {
```

Comment:
Is this safe? change is probably also called when one of the others is **un**checked which would disable paypal in the wrong moment?

Reply:
`.change()` is only firing for the element that is being selected. As an experiment, I added a `console.log()` to both events (line 127 and line 131) - it was only the newly selected item that fired.

The only issue I could see arise is if we have multiple payment providers that do these kinds of shenanigans and en-/disable the pretix-provided payment button at the same time...